src/libraries/System.Net.Connections/src/System/Net/Connections/Connection.cs

```diff
@@ -0,0 +1,288 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Connections
+{
+    /// <summary>
+    /// A connection.
+    /// </summary>
+    public abstract class Connection : ConnectionBase
+    {
+        private Stream? _stream;
+        private IDuplexPipe? _pipe;
+        private bool _initializing;
+
+        /// <summary>
+        /// The connection's <see cref="Stream"/>.
+        /// </summary>
+        public Stream Stream =>
+            _stream != null ? _stream :
+            _pipe != null ? throw new InvalidOperationException("Unable to create Stream; Pipe already initialized.") :
+            (_stream = CreateStream() ?? throw new InvalidOperationException("CreateStream returned null."));
```

Comment:
should this return null instead of throwing?

Reply:
Nope; CreateStream must return non-null.